x-pack/plugins/endpoint/server/routes/alerts.test.ts

```diff
@@ -0,0 +1,191 @@
+    expect(routeConfig.options).toEqual({ authRequired: true });
+    expect(mockResponse.ok).toBeCalled();
+    const alertResultList = mockResponse.ok.mock.calls[0][0]?.body as AlertResultList;
+    expect(alertResultList.total).toEqual(21);
+    expect(alertResultList.request_page_index).toEqual(0);
+    expect(alertResultList.result_from_index).toEqual(0);
+    expect(alertResultList.request_page_size).toEqual(10);
+  });
+
+  it('should return alert results according to pagination params -- POST', async () => {
+    const mockRequest = httpServerMock.createKibanaRequest({
+      method: 'post',
+      body: {
+        page_size: 6,
+        page_index: 3,
+      },
+    });
+    mockScopedClient.callAsCurrentUser.mockImplementationOnce(() => Promise.resolve(data));
+    [routeConfig, routeHandler] = routerMock.post.mock.calls.find(([{ path }]) =>
+      path.startsWith('/api/endpoint/alerts')
+    )!;
+
+    await routeHandler(
+      ({
+        core: {
+          elasticsearch: {
+            dataClient: mockScopedClient,
+          },
+        },
+      } as unknown) as RequestHandlerContext,
+      mockRequest,
+      mockResponse
+    );
+
+    expect(mockScopedClient.callAsCurrentUser).toBeCalled();
+    expect(routeConfig.options).toEqual({ authRequired: true });
+    expect(mockResponse.ok).toBeCalled();
+    const alertResultList = mockResponse.ok.mock.calls[0][0]?.body as AlertResultList;
+    expect(alertResultList.total).toEqual(21);
+    expect(alertResultList.request_page_index).toEqual(3);
+    expect(alertResultList.result_from_index).toEqual(18);
+    expect(alertResultList.request_page_size).toEqual(6);
+  });
+
+  it('should return alert results according to pagination params -- GET', async () => {
+    const mockRequest = httpServerMock.createKibanaRequest({
+      path: '/api/endpoint/alerts',
+      query: {
+        page_size: 3,
+        page_index: 2,
+      },
+    });
+    mockScopedClient.callAsCurrentUser.mockImplementationOnce(() => Promise.resolve(data));
+    [routeConfig, routeHandler] = routerMock.get.mock.calls.find(([{ path }]) =>
+      path.startsWith('/api/endpoint/alerts')
+    )!;
+
+    await routeHandler(
+      ({
+        core: {
+          elasticsearch: {
+            dataClient: mockScopedClient,
+          },
+        },
+      } as unknown) as RequestHandlerContext,
+      mockRequest,
+      mockResponse
+    );
+
+    expect(mockScopedClient.callAsCurrentUser).toBeCalled();
+    expect(routeConfig.options).toEqual({ authRequired: true });
+    expect(mockResponse.ok).toBeCalled();
+    const alertResultList = mockResponse.ok.mock.calls[0][0]?.body as AlertResultList;
+    expect(alertResultList.total).toEqual(21);
+    expect(alertResultList.request_page_index).toEqual(2);
+    expect(alertResultList.result_from_index).toEqual(6);
+    expect(alertResultList.request_page_size).toEqual(3);
+  });
+
+  it('should correctly validate params', async () => {
```

Comment:
I don't think that this is the right approach long term, but we should merge this and come back to it once we have a better understanding of how to test end to end.

Reply:
Good place for a FIXME or TODO comment and new issue?

Reply:
@crowens We need some basic unit tests that covers core behaviors and part of that is using mocks and asserting on queries that are built and how methods are called. The e2e tests for api lives here 
https://github.com/elastic/kibana/tree/master/x-pack/test/api_integration and the broader types of test e.g. functional etc https://github.com/elastic/kibana/tree/master/x-pack/test. These tests requires a running server, and I think we need both. On the EMT we always add both test on every PR.

Reply:
@crowens Yep. I'll create a ticket. We'll be revisiting pagination in an upcoming PR, so that will be a good time to address it. A PR just went in for Endpoint List that took a similar approach to testing, so we should address it there as well.

Reply:
Oooh let me take a look.

Reply:
I'm gonna write an integration test and then we can re-review.

Reply:
Functional tests added.